src/common.rs

```diff
@@ -1,28 +1,59 @@
 use std::str::Utf8Error;
 
-use datafusion::arrow::array::{Array, ArrayRef, Int64Array, LargeStringArray, StringArray, UInt64Array};
+use datafusion::arrow::array::{
+    Array, ArrayRef, AsArray, Int64Array, LargeStringArray, StringArray, StringViewArray, UInt64Array,
+};
 use datafusion::arrow::datatypes::DataType;
 use datafusion::common::{exec_err, plan_err, Result as DataFusionResult, ScalarValue};
 use datafusion::logical_expr::ColumnarValue;
 use jiter::{Jiter, JiterError, Peek};
 
 use crate::common_union::{is_json_union, json_from_union_scalar, nested_json_array};
 
-pub fn check_args(args: &[DataType], fn_name: &str) -> DataFusionResult<()> {
+/// check the args of a function return an error, return the return type
```

Comment:
I think we could have a better docstring with `Arguments` etc

Reply:
#41